### DISCUSSION TITLE
Not Able to delete message from SQS

### DISCUSSION THREAD

**Discussion by Rajneesh-Testsigma** [aws/aws-sdk-java-v2#3304] [Q&A]
I have one FIFO sqs ... I am trying to delete the message from it but getting this kind of error : 

software.amazon.awssdk.core.exception.SdkClientException: Unable to execute HTTP request: Acquire operation took longer than the configured maximum time. This indicates that a request cannot get a connection from the pool within the specified maximum time. This can be due to high request rate.
Consider taking any of the following actions to mitigate the issue: increase max connections, increase acquire timeout, or slowing the request rate.

Increasing the max connections can increase client throughput (unless the network interface is already fully utilized), but can eventually start to hit operation system limitations on the number of file descriptors used by the process. If you already are fully utilizing your network interface or cannot further increase your connection count, increasing the acquire timeout gives extra time for requests to acquire a connection before timing out. If the connections doesn't free up, the subsequent requests will still timeout.
If the above mechanisms are not able to fix the issue, try smoothing out your requests so that large traffic bursts cannot overload the client, being more efficient with the number of times you need to call AWS, or by increasing the number of hosts sending requests.

Client for deleting I have created like this. : 

 return SqsAsyncClient.builder()
      .endpointOverride(URI.create("http://localhost:4566"))
      .region(Region.of(region))
      .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create(awsAccessKey, awsSecretKey)))
      .build();

**Comment by debora-ito** [comment]
The message indicates that the acquire operation (get a connection from the connection pool) timed out.

How often are you seeing the message?
Enabling the client-side metrics can help giving insights on the number of active, available and pending connections, and how long the requests are taking. See here for instructions in how to enable - https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/metrics.html
